### DISCUSSION TITLE
Can TonexOneController handle 2 Tonex one units....

### DISCUSSION THREAD

**Discussion by waltiti17** [Builty/TonexOneController#9] [Ideas]
Hey Greg Already left some comments on your YouTube videos and facebook post.
First thanks a lot, a lot.

Second, some suggestions.

Is possible to handle 2 tonex one units, not at the same time, maybe with a physical switch or software switch in the TonexOneController software?

is possible to use on the big Tonex Pedal, I think is a complete diferent unit, but maybe.

Thanks a lot again

**Comment by Builty** [comment]
Hi,
2 Tonex's would be tricky. The version of ESP IDF I'm using currently doesn't support USB hubs, so it would need this kind of thing:
https://www.lindy.com.au/2-port-usb-3-2-gen-2-type-c-bi-directional-switch

Expensive though. I'm also not confident that the current code would handle change over nicely. It would probably need some better handling of disconects and clean up ready for the next one.

Espressif have recently added USB hub support, so thats probably a better way to go. I haven't tried to use a new IDF myself yet, but I probably will look at this soon.
If a USB hub was used, both could connect at once and then its more of a firmware issue to solve.

With the bigger Tonex, I have no idea. I don't own one myself. I'd guess its probably similar, but the messaging for the One has preset banks A,B and C, as it only has 3 banks selectable normally (A/B toggle and C for stomp mode.)
The bigger Tonex doesn't have that limitation so I'd guess the messages are different.
It already has midi so the only real gain would be the LCD I guess.

**Comment by eykh** [comment]
I follow the idea of a small headless (ESP32 S3 Zero, ~10$ board) TonexOneController. If this works (Builty/Greg supports me) AND if wired MIDI works (I guess Greg will support this as well), you could have a TonexOneController for each of the Tone Ones and send the required MIDI PC commands from any wired MIDI source/controller.

I'll keep on testing and fiddling...
